daemon/algod/api/server/v2/account.go

```diff
@@ -183,33 +183,58 @@ func AccountToAccountData(a *generated.Account) (basics.AccountData, error) {
 	var assetParams map[basics.AssetIndex]basics.AssetParams
 	if a.CreatedAssets != nil && len(*a.CreatedAssets) > 0 {
 		assetParams = make(map[basics.AssetIndex]basics.AssetParams, len(*a.CreatedAssets))
+		var err error
 		for _, ca := range *a.CreatedAssets {
 			var metadataHash [32]byte
-			copy(metadataHash[:], *ca.Params.MetadataHash)
-			manager, err := basics.UnmarshalChecksumAddress(*ca.Params.Manager)
-			if err != nil {
-				return basics.AccountData{}, err
+			if ca.Params.MetadataHash != nil {
+				copy(metadataHash[:], *ca.Params.MetadataHash)
 			}
-			reserve, err := basics.UnmarshalChecksumAddress(*ca.Params.Reserve)
-			if err != nil {
-				return basics.AccountData{}, err
+			var manager, reserve, freeze, clawback basics.Address
+			if ca.Params.Manager != nil {
+				if manager, err = basics.UnmarshalChecksumAddress(*ca.Params.Manager); err != nil {
+					return basics.AccountData{}, err
+				}
```

Comment:
I'm surprised the addresses need to be copied. I would think that the `basics.UnmarshalChecksumAddress` would allocate storage when unmarshelling these.

Reply:
the problem is `ca.Params.Manager` can be `nil`

Reply:
I see. So you just fixed unrelated issue.. ok.

Reply:
yeah, this `AccountToAccountData` func is only used by `dryrun` and `tealdbg`